pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -300,18 +300,22 @@ func (f *Fit) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldOb
 	return framework.Queue, nil
 }
 
-// isResourceScaleDown checks whether the resource request of the modified pod is less than the original pod
-// for the resources requested by the pod we are trying to schedule.
-func (f *Fit) isResourceScaleDown(targetPod, originalOtherPod, modifiedOtherPod *v1.Pod) bool {
-	if modifiedOtherPod.Spec.NodeName == "" {
-		// no resource is freed up whatever the pod is modified.
+// isResourceScaleDown checks whether an update event may make the pod schedulable. Specifically:
+// - Returns true when an update event shows a scheduled pod's resource request got reduced.
+// - Returns true when an update event is for the unscheduled pod itself, and it shows the pod's resource request got reduced.
+func (f *Fit) isResourceScaleDown(targetPod, originalPod, modifiedPod *v1.Pod) bool {
+	samePod := modifiedPod.UID == targetPod.UID
+	scheduledPod := modifiedPod.Spec.NodeName != ""
+	if !samePod && !scheduledPod {
```

Comment:
From the comment on L310, a reader can understand what each comparison means. 

```suggestion
	if modifiedPod.UID != targetPod.UID && modifiedPod.Spec.NodeName == "" {
```